src/ray/raylet/raylet.cc

```diff
@@ -100,11 +100,13 @@ Raylet::Raylet(instrumented_io_context &main_service,
 
   // Setting up autoscaler related fields from ENV
   auto instance_id = std::getenv(kNodeCloudInstanceIdEnv);
-  self_node_info_.set_instance_id(instance_id ? instance_id : "");
```

Comment:
I think the old way is actually preferred for checking whether pointer is nullptr: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es87-dont-add-redundant--or--to-conditions